autogpt/agent/agent.py

```diff
@@ -166,7 +166,11 @@ def signal_handler(signum, frame):
                         assistant_reply_json, assistant_reply, self.config
                     )
                     if self.config.speak_mode:
-                        say_text(f"I want to execute {command_name}")
+                        command_vars = vars(
+                            self.command_registry.commands[command_name]
+                        )
+                        description = command_vars["description"]
```

Comment:
```suggestion
                        command = self.command_registry.get_command(command_name)
                        if command is not None:
                            say_text(command_name, self.config, description)
```

Reply:
Now that I think of it, it's very strange that we don't check whether a command exists before asking for approval to execute it.